src/Tizen.Applications.ComponentBased/Tizen.Applications.ComponentBased.Common/ComponentFactoryBase.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Tizen.Applications.CoreBackend;
+using static Interop.CBApplication;
+using static Tizen.Applications.CoreBackend.DefaultCoreBackend;
+
+namespace Tizen.Applications.ComponentBased.Common
+{
+    internal abstract class ComponentFactoryBase
+    {
+        public IList<BaseComponent> ComponentInstances { get; }
```

Comment:
Do not directly expose List

if you want to provide way to adding on the ComponentInstances, 

You can provide a method to add
``` c#
  private IList<BaseComponent> ComponentInstances { get; }
  public IEnumerable<BaseComponent> Instances => ComponentInstances;
  protected void AddComponent(BaseComponent comp) {
      ComponentInstances.Add(comp);  
  }

  protected void RemoveComponent(BaseComponent comp) {
      ComponentInstances.Remove(comp);  
  }
```

Reply:
Done